File: microservice/config/src/main/java/com/michibaum/config/ConfigApplication.java
package com.michibaum.config;

import brave.sampler.Sampler;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.config.server.EnableConfigServer;
import org.springframework.cloud.context.config.annotation.RefreshScope;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableScheduling;

@SpringBootApplication
@EnableConfigServer
@RefreshScope
@EnableScheduling
public class ConfigApplication {

    public static void main(String[] args) {
        SpringApplication.run(ConfigApplication.class, args);
    }

    @Bean
    public Sampler defaultSampler(){
        return Sampler.ALWAYS_SAMPLE;
    }

}

File: microservice/config/src/main/kotlin/com/michibaum/config/ConfigApplication.kt
package com.michibaum.config

import brave.sampler.Sampler
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.config.server.EnableConfigServer
import org.springframework.cloud.context.config.annotation.RefreshScope
import org.springframework.context.annotation.Bean
import org.springframework.scheduling.annotation.EnableScheduling


@SpringBootApplication
@EnableConfigServer
@RefreshScope
@EnableScheduling
class ConfigApplication {

	fun main(args: Array<String>) {
		runApplication<ConfigApplication>(*args)
	}

	@Bean
	fun defaultSampler(): Sampler = Sampler.ALWAYS_SAMPLE

}


